python/lsst/ts/standardscripts/maintel/set_dof.py

```diff
@@ -42,11 +47,131 @@ class SetDOF(ApplyDOF):
 
     """
 
+    @classmethod
+    def get_schema(cls):
+        schema_yaml = """
+            $schema: http://json-schema.org/draft-07/schema#
+            $id: https://github.com/lsst-ts/ts_standardscripts/maintel/SetDOF.yaml
+            title: SetDOF v1
+            description: Configuration for SetDOF.
+            type: object
+            properties:
+              day:
+                description: Day obs to be used for synchronizing the state.
+                type: number
+                default: null
+              seq:
+                description: Sequence number to be used for synchronizing the state.
+                type: number
+                default: null
+            anyOf:
+                - required:
+                    - day
+                    - seq
+                - required:
+                    - dofs
+            additionalProperties: false
+        """
+        schema_dict = yaml.safe_load(schema_yaml)
+
+        base_schema_dict = super(ApplyDOF, cls).get_schema()
+
+        for prop in base_schema_dict["properties"]:
+            schema_dict["properties"][prop] = base_schema_dict["properties"][prop]
+
+        return schema_dict
+
+    async def configure(self, config) -> None:
+        """Configure script.
+
+        Parameters
+        ----------
+        config : `types.SimpleNamespace`
+            Script configuration, as defined by `schema`.
+        """
+        super(ApplyDOF, self).configure(config)
+
+        if hasattr(config, "day") and hasattr(config, "seq"):
+            self.day = config.day
+            self.seq = config.seq
+
+    async def get_image_time(self, day, seq):
+        """Get the image time from the given day and sequence number.
+
+        Parameters
+        ----------
+        day : `int`
+            Day obs to be used for synchronizing the state.
+        seq : `int`
+            Sequence number to be used for synchronizing the state.
+
+        Returns
+        -------
+        image_time : `datetime.datetime`
+            Image time.
+        """
+
+        query = f"""
+            SELECT  "imageDate", "imageNumber"
+            FROM "efd"."autogen"."lsst.sal.CCCamera.logevent_endReadout"
+            WHERE imageDate =~ /{day}/ AND imageNumber = {seq}
+        """
+
+        time = await self.client.influx_client.query(query)
```

Comment:
I think you need to make some validation here that it returns something... what if the users passed a bad image number?